linux/keyman-config/keyman_config/ibus_util.py

```diff
@@ -30,6 +30,7 @@ def install_to_ibus(bus, keyboard_id):
         preload_engines = ibus_settings.get_strv("preload-engines")
         logging.debug(preload_engines)
         if keyboard_id not in preload_engines:
+            preload_engines.clear()
```

Comment:
I added the clear() because preload_engines was getting
```
BDH/BDH.kmx
bdh/BDH.kmx
```

Reply:
I removed the `.clear()`

Reply:
in that case it would be better to uninstall_from_ibus the BDH.kmx as well as bdh.kmx if that is the filename in kmp.json

If needed you (i.e. a developer) can manually read and write preload_engines using dconf

`dconf read /desktop/ibus/general/preload-engines`
`dconf write /desktop/ibus/general/preload-engines <engine list>`

If there are upper case keyboard ids or package ids in the wild then I don't think this all handles that yet, but hopefully they aren't posssible

Reply:
weird - now I'm not seeing those spurious keyboards in preload_engines. I'll just add a comment in case it shows up again